src/modules/75-cd/components/PipelineStudio/DeployEnvSpecifications/DeployEnvSpecifications.tsx

```diff
@@ -101,7 +101,12 @@ export default function DeployEnvSpecifications(
         if (specObject) {
           // istanbul ignore else
           if (value.environment) {
-            specObject.environment = value.environment
+            specObject.environment = {
+              ...value.environment,
+              provisioner:
+                (value.environment.provisioner && stage?.stage?.spec?.environment?.provisioner) ||
```

Comment:
Above code:  `( a && b) || a` is not clear. Can it be simplified to something else or at-least put some comments.

Reply:
case (a && b) || a

a = undefined
b = 1
(a && b) || a
==> undefined (a)
'expected behavior'
-------------------
a = 1
b = 2
==> 2 (b)
(a && b) || a
'expected behavior'
-------------------
a = 1
b = undefined
(a && b) || a
===> 1
'expected behavior'


case a || b

a = 1
b = undefined

=>> 1 (a) 'expected behavior'
------------------------
a = 1
b = 2
=>> 1 (a)'expected behavior'
------------------------
a = undefined
b = 1
=>> 1 (b) 'unexpected behavior'

Reply:
Thanks @litvinovstas for explanation, using same explanation we can make code more readable. 
Also, if possible we can write 1 -2 lines of inline code comments why and in which flows these use cases are coming.

```suggestion
                const defaultProvisioner = value.environment.provisioner
                provisioner : defaultTo(defaultProvisioner && stage?.stage?.spec?.environment?.provisioner, defaultProvisioner)
                
```